File: solutions/543.diameter-of-binary-tree/diameter-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int res = 0;
    public int diameterOfBinaryTree(TreeNode root) {
        getDeepth(root);
        return res;
    }
    
    
    public int getDeepth(TreeNode node){
        if(node == null) return 0;
        int l = getDeepth(node.left);
        int r = getDeepth(node.right);
        res = Math.max(l+r, res);
        return 1 + Math.max(l, r);
    }
}

File: solutions/543.diameter-of-binary-tree/diameter-of-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var diameterOfBinaryTree = function(root) {
    if(root === null) return 0;
    let res = 0;
    helper(root);
    return res;
    
    function helper(node){
        if(node === null) return 0;
        const left = helper(node.left);
        const right = helper(node.right);
        res = Math.max(res, left + right);
        return 1 + Math.max(left, right);
    }
    
    
};